lib/ruby_lsp/requests/code_action_resolve.rb

```diff
@@ -310,24 +367,18 @@ def switch_block_body(body, indentation)
             start: { line: body_loc.start_line - 1, character: body_loc.start_column },
             end: { line: body_loc.end_line - 1, character: body_loc.end_column },
           },
-          node_types: [Prism::BlockNode],
+          node_types: [Prism::BlockNode, Prism::HashNode, Prism::ArrayNode],
```

Comment:
We need to locate only the top level block, then descend into the body and start correcting all statements inside.

With this approach, you could end up finding an array node mixed in a bunch of other statements and then you'd miss everything else. For example, I wouldn't expect it to work for this case:

```ruby
[].each do |a, b, c|
  a = []
  { b: [a] }
end
```